src/sql/src/session/hint.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal::Private;
+
+/// Prometheus label for [`ApplicationNameHint::Unspecified`].
+const UNSPECIFIED_LABEL: &str = "unspecified";
+/// Prometheus label for [`ApplicationNameHint::Unrecognized`].
+const UNRECOGNIZED_LABEL: &str = "unrecognized";
+/// Prometheus label for [`ApplicationNameHint::Psql`].
+const PSQL_LABEL: &str = "psql";
+/// Prometheus label for [`ApplicationNameHint::Dbt`].
+const DBT_LABEL: &str = "dbt";
+/// Prometheus label for [`ApplicationNameHint::WebConsole`].
+const WEB_CONSOLE_LABEL: &str = "web_console";
+
+/// A hint for what application is making a request to the adapter.
+///
+/// Note: [`ApplicationNameHint`] gets logged as a label in our Prometheus metrics, and for
+/// labels we need to be conscious of the cardinality, so please be careful with how many
+/// variants we add to this enum.
+///
+/// Note: each enum variant contains an `internal::Private` to prevent creating this enum
+/// directly. To create an instance of [`ApplicationNameHint`] please see
+/// [`ApplicationNameHint::from_str`].
+#[derive(Debug, Copy, Clone)]
+pub enum ApplicationNameHint {
+    /// No `application_name` was set.
+    Unspecified(Private),
+    /// An `application_name` was set, but it's not one we recognize.
+    Unrecognized(Private),
+    /// Request came from `psql`.
+    Psql(Private),
+    /// Request came from `dbt`.
+    Dbt(Private),
+    /// Request came from our web console.
+    WebConsole(Private),
+}
+
+impl ApplicationNameHint {
+    pub fn from_str(s: Option<&str>) -> Self {
```

Comment:
I don't think this can ever be none because it's must always exist in the session, but defaults to the empty string.

Reply:
Yeah this is an artifact of a previous impl, updated!